plugins/FirmwareUpdateChecker/FirmwareUpdateCheckerLookup.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+import json
+
+from typing import Callable, Dict, List, Optional
+
+from UM.Logger import Logger
+from UM.Version import Version
+
+from UM.i18n import i18nCatalog
+i18n_catalog = i18nCatalog("cura")
+
+
+def get_settings_key_for_machine(machine_id: int) -> str:
+    return "info/latest_checked_firmware_for_{0}".format(machine_id)
+
+
+def default_parse_version_response(response: str) -> Version:
+    raw_str = response.split("\n", 1)[0].rstrip()
+    return Version(raw_str.split("."))  # Split it into a list; the default parsing of "single string" is different.
+
+
+class FirmwareUpdateCheckerLookup:
+    JSON_NAME_TO_VERSION_PARSE_FUNCTION = {"default": default_parse_version_response}
```

Comment:
So far, the whole business of the version parse functions is unnecessary since they all use the default parser. I think we should leave this unconfigurable by the printer, to be lean.

This would mean that a Cura update is necessary if this ever needs to become different for certain printers. But currently that is needed anyway because they would need to add their function to this dictionary in the source code.

Reply:
Would it be worthwhile to move this information to the printer definition metadata instead of putting it in the plugin, like my (usb) firmware updater rework does? Then it becomes more open to 3rd parties to add checks, without having to hardcode these into Cura too.

Reply:
Lean it is. I've removed the (currently) superfluous parse-function functionality.

@fieldOfView I've put the other meta-data in the actual printer-definition now (see latest commit as of writing), but I've removed the changeable-parse-function-functionality per @Ghostkeeper s request for now. As far as I'm concerned, if you know a good way to re-add them without hardcoding stuff, please go ahead :-)